libraries/lib-wave-track/CachingPlayableSequence.cpp

```diff
@@ -0,0 +1,115 @@
+#include "CachingPlayableSequence.h"
+#include "AudioSegmentSampleView.h"
+#include "WaveTrack.h"
+
+#include <cassert>
+
+namespace
+{
+void FillBufferFromTrackBlockSequence(
+   const TrackSampleView& track, float* buffer, size_t bufferSize)
+{
+   size_t written = 0;
+   for (const auto& segment : track)
+   {
+      size_t toWrite = std::min(
+         segment.GetSampleCount().as_long_long(),
+         static_cast<long long>(bufferSize - written));
+      written += segment.Copy(buffer, toWrite);
+   }
+   std::fill(buffer + written, buffer + bufferSize, 0.f);
+}
+} // namespace
+
+CachingPlayableSequence::CachingPlayableSequence(const WaveTrack& waveTrack)
+    : mWaveTrack(waveTrack)
```

Comment:
`{}` member initializer syntax?

Reply:
Right, I never took the habit for some reason.

Reply:
You should also remember, though, that sometimes you must use `()`.  One example is initializing a vector with a size.  If the element type is implicitly constructible from an int or size_t, the `{}` syntax will invoke the wrong constructor.

This danger exists when a class has a constructor taking `std::initializer_list`.